src/polyfillstub/intersection-observer-stub.js

```diff
@@ -60,9 +73,25 @@ export function upgradePolyfill(win, installer) {
   // instance deployed in v0.js vs the polyfill extension.
   const Stub = /** @type {typeof IntersectionObserverStub} */ (win.IntersectionObserver);
   if (Stub && UPGRADERS in Stub) {
+    const NativeIntersectionObserver =
```

Comment:
The `if` above for `if (UPGRADES in Stub)` will be completely missed from the `!supportsDocumentRoot` branch. So I think you'd still want to use `InObStub` class. Alternative to your code could be using `InObStub` class for all unsupported cases, but return the native intersection observer from its constructor when conditions meet. In that case you'd probably want to go back to your idea of using a `.native` prop. It could go right next to the `UPGRADES` prop.